bench/layers.js

```diff
@@ -55,6 +54,8 @@ async function main() {
       for (let j = 0; j < RUNS_PER_TIER; j += 1) {
         runs.push(await start(current.fn, layers));
       }
+      // Give cellx time to release its global pendingCells array
+      await new Promise((resolve) => setTimeout(resolve, 0));
```

Comment:
Ooo thanks heaps for cathing this @jviide! 

Do we still need this since you added `dispose` calls for cellx? Could we also consider running cellx last to prevent it from leaking over and potentially affecting other tests? I wasn't aware it was such a memory hog as I adapted this bench from a sandbox and I didn't bother looking too deep. I'll definitely be doing memory profiles in the future, is there a tool you use for this?

Reply:
The disposal part makes the size of accumulated chunk of memory smaller, and the timeout thing lets that chunk be released between benchmarks.

My "process" was very haphazard 😀 First I just tried reordering / disabling benchmarks for the different libraries to confirm that there's something funky going on. Then I suspected it's memory related due to Node running out of memory, so I launched Node with the `--expose-gc` option and inserted `global.gc()` calls after each library benchmark. Also the `--trace_gc` option was interesting.

For a better view I started Node.js with the `--inspect` option and connected the Chrome inspector to it (described e.g. here: https://sematext.com/blog/nodejs-memory-leaks/#toc-v8-inspector-chrome-dev-tools-20). I had already placed a `debugger;` command to `main` just after the tests, otherwise Node would have exited `main` and garbage collected the memory before I had a chance to take Heap snapshot in the inspector. After taking the snapshot I sorted the view by "retained size", and noticed that pendingCells was indirectly holding to large amounts of memory. It should be noted that I had to disable the two last test tiers before taking the Heap snapshot, because otherwise it'd taken forever to process.

I wonder if running each library benchmark in its own [worker thread](https://nodejs.org/api/worker_threads.html) would help to isolate these kinds of side-effects... Though implementing it is probably a whole another can of worms 🙂

Reply:
Thank you so much for the debugging tips! Sounds fun, I'll definitely be trying it all out. I'm still a noob on that front 😄

The workers idea is awesome but to avoid a time sink right now we can stick with these fixes. I'll follow up with:

- Move `cellx` last to prevent it messing with other benchs.
- Have a separate `batch` benchmark to see how libs perform in a more fair setting.
- Update benchmark screenshots in README

Future considerations:

- Worker threads for bench runs
- Memory profiling benchs - I wonder if there's a way to automate this using something like Puppeteer/Playwright.

If you have any other suggestions please let me know and thank you again for this PR. If I end up attempting any "future considerations" I'll give you a ping.